lib/pbio/src/os.c

```diff
@@ -69,11 +77,27 @@ void pbio_os_start_process(pbio_os_process_t *process, pbio_os_process_func_t fu
     }
 
     // Initialize the process.
-    process->func = func;
-    process->context = context;
     process->next = NULL;
+    process->context = context;
+
+    pbio_os_process_reset(process, func);
+}
+
+/**
+ * Resets an existing process to the initial state with a new function and context.
```

Comment:
Resetting a process sounds dangerous without some kind of cleanup/teardown of the current state. Might be worth mentioning that in the description.